vterm-module-make.el

```diff
@@ -5,11 +5,54 @@
 (defvar vterm-install-buffer-name " *Install vterm"
   "Name of the buffer used for compiling vterm-module.")
 
+(defun vterm-module-compilation-requirements-satisfied ()
+  "Return t if the requirements for the successful compilation are satisfied.
+
+This means checking if cmake is available, if it is newer than
+version 3.11.0, and if libtool is available."
+
+  ;; Check if cmake is available
+  (unless (executable-find "cmake")
+    (error "Vterm needs to be compiled but a dependency, cmake, is missing"))
+
+  ;; Now we can assume cmake is available, but is it new enough?
+  ;; We need a minimum version, at the moment 3.11.0, we check this
+  ;; by invoking cmake --version and matching the number inside
+  ;; At the moment, the output of this command looks like
+  ;;
+  ;; cmake version 3.14.6
+  ;;
+  ;; CMake suite maintained and supported by Kitware (kitware.com/cmake).
+  ;;
+  ;; So, we extract the version from the first line.
+  (let* ((cmake-min-required-version "3.11.0")
+        (cmake-version-output (shell-command-to-string "cmake --version"))
+        (cmake-version
+         (progn
+           (string-match "^cmake version \\([0-9.]+\\)" cmake-version-output)
+           (match-string 1 cmake-version-output))))
+    (when
+        (string< cmake-version cmake-min-required-version)
```

Comment:
Just a note, use `version<` when comparing versions. `(string< "3.0.0" "10.0.0")` is false.

Reply:
> My point is that cmake is the build tool, and is the canonical place to declare build dependencies. Otherwise the build tool will become a cmake+elisp hybrid, which is unnecessary.
> 
> I agree that checking if cmake is available is a good idea, but checking its version already works, and does not need to be duplicated.
> 
> Regarding libtool, I think this is the proper way to do it in cmake (I'm no cmake expert):
> 
> ```
> find_program(LIBTOOL NAMES libtool glibtool)
> if(NOT LIBTOOL)
>   message(FATAL_ERROR "libtool must be installed")
> endif()
> ```
> 
> I don't really see how errors are intimidating if done like this.
> 
> Also, I think that maybe libtool check is only necessary if `USE_SYSTEM_LIBVTERM=false`, i.e. in the branch with ExternalProject_add.

Thanks, I am happy with your suggestions. I think this is a good way to handle this.